extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/ConstraintMappingProvider.java

```diff
@@ -0,0 +1,25 @@
+package io.quarkus.hibernate.validator.runtime;
+
+import javax.validation.Validator;
+
+import org.hibernate.validator.BaseHibernateValidatorConfiguration;
+
+/**
+ * Meant to be implemented by a CDI bean that provides programmatic constraint mapping for the default
+ * {@link javax.validation.Validator}.
+ * <p>
+ * All implementations (that are registered as CDI beans) are taken into account when producing the default
+ * {@link Validator}.
+ * <p>
+ */
+public interface ConstraintMappingProvider {
```

Comment:
Given you can basically do anything you want on the configuration, I wonder if we should name it `HibernateValidatorFactoryCustomizer` and make the mechanism similar to what we did for `ObjectMapperCustomizer`.

I have a doubt about how we should manage the priority though as `ObjectMapperCustomizer` has a `getPriority()` method but the way you did things with `Arc.container().listAll()` should respect the `@Priority` annotations already so we wouldn't need this stuff.

I asked @geoand on Zulip and I pinged you there.